### PR TITLE
Conv weight fix

### PR DESCRIPTION
**Pull request by kahmed10**
Includes changes to broadcasting a scalar.

### CODE REVIEW

File: src/include/migraphx/op/broadcast.hpp [hunk 1]
```diff
@@ -49,8 +49,6 @@ struct broadcast
         if(std::all_of(
                broadcast_lens.cbegin(), broadcast_lens.cend(), [&](auto x) { return x == 1; }))
         {
-            if(axis != 0)
-                MIGRAPHX_THROW("BROADCAST: when broadcasting tensor of size 1, axis should be 0");
```
Comment (pfultz2) [inline src/include/migraphx/op/broadcast.hpp:53]: Can you add a unit test for this?

Comment (kahmed10) [reply src/include/migraphx/op/broadcast.hpp:53]: We have unit tests for this if statement under `op_shape_test.cpp`. I also added a scalar test for broadcast in tf.

Comment (pfultz2) [reply src/include/migraphx/op/broadcast.hpp:53]: There isnt a test for non zero axis in op_shape_test.cpp, only for when axis is 0.

Comment (kahmed10) [reply src/include/migraphx/op/broadcast.hpp:53]: Added

**Review by pfultz2 — APPROVED** [review]

File: test/op_shape_test.cpp [hunk 2]
```diff
@@ -346,6 +346,15 @@ TEST_CASE(broadcast)
...
+        std::vector<std::size_t> lens{1, 1};
+        migraphx::shape input{migraphx::shape::float_type, {1}, {0}};
+        expect_shape(migraphx::shape{migraphx::shape::float_type, {1, 1}, {0, 0}},
+                     migraphx::op::broadcast{1, lens},
+                     input);
+    }
+
     {
         std::vector<std::size_t> lens{1, 1};
         migraphx::shape input{migraphx::shape::float_type, {4, 1, 3}};
```
Comment (pfultz2) [inline test/op_shape_test.cpp:360]: This test should be removed as it fails, since it no longer throws.

File: test/tf/gen_tf_pb.py [hunk 3]
```diff
@@ -145,6 +145,14 @@ def biasadd_test(g1):
         tf.nn.bias_add(g1_input, g2_input, name='bias_add1')
 
 
+@tf_test
+def biasadd_scalar_test(g1):
```
Comment (pfultz2) [inline test/tf/gen_tf_pb.py:149]: I think you forgot to commit these new files.

File: test/op_shape_test.cpp (hunk 2 above)
Comment (kahmed10) [reply test/op_shape_test.cpp]: removed

File: test/tf/gen_tf_pb.py (hunk 3 above)
Comment (kahmed10) [reply test/tf/gen_tf_pb.py:149]: added

File: src/include/migraphx/op/broadcast.hpp (hunk 1 above)
Comment (scxiao) [reply src/include/migraphx/op/broadcast.hpp:53]: We may need to add a check that input either is a scalar or contains only one element? Then the shape unit test there can still pass.

**Comment by pfultz2** [comment]
Can you merge this in first since it fixes a serious bug? And then we can open an issue to improve error checking in the broadcast operators?

**Comment by mvermeulen** [comment]
@scxiao  is this ok to approve?

**Comment by scxiao** [comment]
> @scxiao is this ok to approve?

If we use another PR for the checking. I am fine to approve this one.

**Review by scxiao — APPROVED** [review]
